doc/design.md

```diff
@@ -0,0 +1,94 @@
+# Design
+
+## DPPP: a framework for streaming radio interferometric pipelines
+DPPP (DP3, short for Default PreProcessing Pipeline) is a framework
+for efficiently processing time-ordered visibilities,
+e.g. freshly correlated data. It was developed for the LOFAR
+telescope, and is used for preprocessing its imaging pipelines.
+DPPP user documentation is found at the [LOFAR Wiki](https://goo.gl/2uAaN9).
+
+When DPPP reads a chunk of data, it calls all the steps to do
+all processing possible on this data. The last step will then
+write out the processed data. In this way, the data is read
+and written only once.
+
+![High level overview of DPPP](doc/images/diagram.png)
+
+DPPP runs a pipeline in clearly defined stages, to make spotting
+errors as easy as possible.
+* **Initialise** This connects all the processing steps together. This step is data independent.
+* **Tune At** this stage only metadata, such as the number of channels, is read, and steps are adjusted to this.
+* **Process** Data is piped through the steps in buffers of time. The buffer size is defined by the steps.
+
+## Measurement Set standard
+DPPP adheres to the Measurement Set (MS) standard, so it
+can be used with data from all radio telescopes. DPPP can
+also create a new MS. When an appropriate input step is in
+place (e.g. a streaming connection to a correlator), MSs can
+be written from scratch.
+
+DPPP supports only regularly shaped MSs, meaning that every
+time slot must have the same number of baselines and antennas.
+Only one spectral window can be used.
+
+## Built-in steps
+DPPP contains built-in steps that are linked together and process one time slot at a time and is therefore suitable for large measurement sets.
+All the built-in steps derive from the [DPStep](@ref DP3::DPPP::DPStep) class.
```

Comment:
You could add "Its inheritance diagram shows all step types, including the built-in steps."